src/hooks/useDataSource.tsx

```diff
@@ -0,0 +1,23 @@
+import { useEffect, useState } from 'react';
+import { getControlTemplate, getDataSource, OHRIFormField } from '..';
+
+export function useDataSource(field: OHRIFormField) {
+  const [config, setConfig] = useState({});
+  const [dataSource, setDataSource] = useState(null);
+
+  useEffect(() => {
+    if (field.questionOptions.datasource?.id) {
+      setDataSource(getDataSource(field.questionOptions.datasource.id));
+      setConfig(field.questionOptions.datasource.config);
+    } else {
+      const template = getControlTemplate(field.questionOptions.rendering);
+      setDataSource(getDataSource(template.datasource.id));
```

Comment:
What happens if the `template` is undefined?

Reply:
>What happens if the template is undefined?

At this point, we need to throw an error because no template was found and data source was configured.